Sources/Core/ObjectiveCDictionaryExtension.swift

```diff
@@ -0,0 +1,133 @@
+        let propIVarName = "_\(param.snakeCaseToPropertyName())"
+        switch schema {
+        case .boolean:
+            return "[dict setObject: [NSNumber numberWithBool:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+
+        case .float:
+            return "[dict setObject: [NSNumber numberWithFloat:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .integer:
+            return "[dict setObject: [NSNumber numberWithInteger:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .object(_):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: "+propIVarName + " forKey: @\"" + param + "\" ];"
+                ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                ]
+                })
+                ]}
+        case .string(format: .none),
+             .string(format: .some(.email)),
+             .string(format: .some(.hostname)),
+             .string(format: .some(.ipv4)),
+             .string(format: .some(.ipv6)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: "+propIVarName + " forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .string(format: .some(.uri)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: ["+propIVarName + " absoluteString] forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .string(format: .some(.dateTime)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: [[NSValueTransformer valueTransformerForName:\(dateValueTransformerKey)] transformedValue:\(propIVarName)]  forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .enumT(.integer(_)):
+            return "[dict setObject: [NSNumber numberWithInteger:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .enumT(.string(_)):
+            return "[dict setObject: "+enumToStringMethodName(propertyName: param, className: self.className) + "(\(propIVarName))" + " forKey: @\"" + param + "\" ];"
+        case .array(itemType: _):            
+            return ObjCIR.scope {[
```

Comment:
done, when received array of references, we put its dictionaryRepresentation into array